explorer/app/components/TableCreator/tableCreator.tsx

```diff
@@ -32,21 +32,81 @@ export const TableCreator = <T extends object>({
   pagination,
   sort,
 }: TableCreatorProps<T>): JSX.Element => {
-  const reactColumns: Column<T>[] = columns.map((columnConfig) => ({
-    Cell: createCell(columnConfig),
-    Header: columnConfig.header,
-    disableSortBy: !columnConfig.sort,
-    id: columnConfig.sort?.sortKey,
+  const defaultColumns = useMemo(
+    () => columns.filter((column) => !column.hiddenColumn),
+    [columns]
+  );
+  const [visibleColumn, setVisibleColumn] = useState(defaultColumns);
```

Comment:
Maybe pluralize the naming here (`visibleColumns` and `setVisibleColumns`) to help indicate it's an array?